src/main/java/com/amazon/opendistroforelasticsearch/sql/request/SqlRequest.java

```diff
@@ -105,4 +109,14 @@ public BoolQueryBuilder checkAndAddFilter(BoolQueryBuilder boolQuery) throws Sql
         }
         return boolQuery;
     }
+
+    public String removeSensitiveData() {
+        String query = "";
+        try {
+            query = QueryDataMask.maskData(this.sql);
```

Comment:
As Peng commented, I think you can either move all mask logic here or just call `maskData()` outside along with `getSql()`.